src/rest_framework_api_key/models.py

```diff
@@ -40,17 +40,19 @@ def create_key(self, **kwargs: typing.Any) -> typing.Tuple["AbstractAPIKey", str
     def get_usable_keys(self) -> models.QuerySet:
         return self.filter(revoked=False)
 
-    def is_valid(self, key: str) -> bool:
+    def get_from_key(self, key: str) -> "AbstractAPIKey":
         prefix, _, _ = key.partition(".")
-
         queryset = self.get_usable_keys()
+        api_key = queryset.get(prefix=prefix)
+        if not api_key.is_valid(key):
+            raise ValidationError("Key is not valid.")
```

Comment:
If we're to follow the API of `.get()`, we should probably raise a `DoesNotExist` exception here instead. (I think `ValidationError` is reserved for form validation in Django, correct?).

Even beyond the API consistency, I think it does make sense to say "nope, no API key for this generated key" in this case, because what really happens is "yes, we have a key for that prefix, but the rest of the key doesn't match, so there's definitely no such key in the DB".

Reply:
That makes sense to me! I left the error message as `Key is not valid`, but if you think that doesn't make sense with `DoesNotExist` we can change it to something like `Prefix exists, but the key is not valid`.

Reply:
I think we're good — probably best to stay broad enough in the error message here. :-)